src/Composer/Package/Cleaner/Cleaner.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace LastCall\DownloadsPlugin\Composer\Package\Cleaner;
+
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\Gitignore;
+
+class Cleaner implements CleanerInterface
+{
+    public function __construct(private array $ignore)
+    {
+    }
+
+    public function clean(string $dir): void
+    {
+        if (empty($this->ignore)) {
+            return;
+        }
```

Comment:
doesn't this mean if I have no files to ignore, don't clear the folder?

Reply:
Yes, if nothing to ignore, don't touch the folder.
If there are something to ignore, delete files that matched the git ignore patterns.

Reply:
Surely that is a bug though? I want to delete everything if there is nothing to ignore would have been what I thought was the right option. Can you tell me why this should skip if there is nothing to ignore?

Reply:
I think the word `ignore` is causing confusion.

> I want to delete everything if there is nothing to ignore

It's the other way around:
- If I want to delete everything, I need to ignore everything .e.g `*`
- If I want to delete all files begin with `test`, I need to ignore `test*`
- If I want to delete folder named `docs`, I need to ignore `docs/`

I'm truly sorry about the confusion. I keep this convention from the original project https://github.com/civicrm/composer-downloads-plugin?tab=readme-ov-file#configuration-properties

Reply:
No this is a great disambiguation, which I'd failed to follow. Approved.